packages/ckeditor5-ui/src/search/text/searchtextview.ts

```diff
@@ -0,0 +1,414 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/search/text/searchtextview
+*/
+
+import { FocusTracker, KeystrokeHandler, type Locale } from '@ckeditor/ckeditor5-utils';
+import View from '../../view';
+import { default as SearchTextQueryView, type SearchTextQueryViewConfig } from './searchtextqueryview';
+import SearchInfoView from '../searchinfoview';
+import SearchResultsView from '../searchresultsview';
+import FocusCycler from '../../focuscycler';
+import { escapeRegExp } from 'lodash-es';
+
+import type FilteredView from '../filteredview';
+import type ViewCollection from '../../viewcollection';
+import type InputBase from '../../input/inputbase';
+import type InputTextView from '../../inputtext/inputtextview';
+
+import '../../../theme/components/search/search.css';
+
+/**
+ * A search component that allows filtering of an arbitrary view based on a search query
+ * specified by the user in a text field.
+ *
+ *```ts
+ * // This view must specify the `filter()` and `focus()` methods.
+ * const filteredView = ...;
+ *
+ * const searchView = new SearchTextView( locale, {
+ * 	searchFieldLabel: 'Search list items',
+ * 	filteredView
+ * } );
+ *
+ * view.render();
+ *
+ * document.body.append( view.element );
+ * ```
+ */
+export default class SearchTextView<
+	TQueryFieldView extends InputBase<HTMLInputElement | HTMLTextAreaElement> = InputTextView
+> extends View {
+	/**
+	 * Tracks information about the DOM focus in the view.
+	 *
+	 * @readonly
+	 */
+	public focusTracker: FocusTracker;
+
+	/**
+	 * An instance of the keystroke handler managing user interaction and accessibility.
+	 *
+	 * @readonly
+	 */
+	public keystrokes: KeystrokeHandler;
+
+	/**
+	 * A view hosting the {@link #filteredView} passed in the configuration and the {@link #infoView}.
+	 */
+	public resultsView: SearchResultsView;
+
+	/**
+	 * The view that is filtered by the search query.
+	 */
+	public filteredView: FilteredView;
+
+	/**
+	 * The view that displays the information about the search results.
+	 */
+	public infoView: View | undefined;
+
+	/**
+	 * The view that allows the user to enter the search query.
+	 */
+	public queryView: SearchTextQueryView<TQueryFieldView>;
+
+	/**
+	 * Controls whether the component is in read-only mode.
+	 *
+	 * @default true
+	 * @observable
+	 */
+	declare public isEnabled: boolean;
+
+	/**
+	 * The number of results found for the current search query. Updated upon the {@link #search} event.
+	 *
+	 * @default 0
+	 * @observable
+	 */
+	declare public resultsCount: number;
+
+	/**
+	 * The number of the items that can be searched in the {@link #filteredView}. Updated upon the {@link #search} event.
+	 *
+	 * @default 0
+	 * @observable
+	 */
+	declare public totalItemsCount: number;
```

Comment:
Shouldn't `totalItemsCount` be refreshed when the items in `filteredView` change instead of on search? Shouldn't we display a proper info if `filteredView` becomes empty separately from processing search requests?

If you wanted to exclude that from the scope to deliver the component faster, let's create a followup issue for this.

Reply:
It makes sense but it would mean `FilteredView` will have to be broader than that. For instance, it will have to expose some (observable) properties such as `totalItemsCount` itself. I don't think we have time to address that now.

Reply:
https://github.com/ckeditor/ckeditor5/issues/15059